engine/src/main/java/org/hibernate/validator/internal/util/logging/Log.java

```diff
@@ -640,4 +640,7 @@ UnexpectedTypeException getNoValidatorFoundForTypeException(String constraintTyp
 	@LogMessage(level = WARN)
 	@Message(id = 192, value = "Couldn't determine Java version from value %1s; Not enabling features requiring Java 8")
 	void unknownJvmVersion(String vmVersionStr);
+
+	@Message(id = 193, value = "%s is configured more than once via the programmatic constraint definition API.")
```

Comment:
It should be illegal do define one constraint more than once across all `ConstraintMapping`s (as it's illegal to do in one more more XML file(s)).